### DISCUSSION TITLE
script配置

### DISCUSSION THREAD

**Discussion by weineo** [libnyanpasu/clash-nyanpasu#173] [Q&A] [answered]
各位大佬，求助clash verge的script代码！我之前在clash for windows上习惯于对订阅文件进行预处理，分别先将服务商自带的策略组和规则清空，然后再添加自定义的策略组和规则。clash的预处理是yaml格式的，但是clash verge只能用添加script配置来实现，我尝试了很久也没成功移植过来，希望有大佬能够伸出援助之手。我实现清空的代码：
    code: |
      module.exports.parse = (raw, { yaml }) => {
        const rawObj = yaml.parse(raw)
        const groups = []
        const rules = []
        return yaml.stringify({ ...rawObj, 'proxy-groups': groups, rules })
      }
当然了，如果有大佬能够顺手翻译一下完整的代码那就更感谢了。
parsers: # array
  - reg: ^.*$ 
  # 下面是删除服务商自带的策略组和规则
    code: |
      module.exports.parse = (raw, { yaml }) => {
        const rawObj = yaml.parse(raw)
        const groups = []
        const rules = []
        return yaml.stringify({ ...rawObj, 'proxy-groups': groups, rules })
      } 

    yaml:
      mix-object:
        mixed-port: 7888
        allow-lan: true
        bind-address: "*"
        mode: rule
        log-level: info
        external-controller: 127.0.0.1:9090
        dns:
          enable: true
          ipv6: true
          default-nameserver:
            - 223.5.5.5
            - 119.29.29.29
            - 192.168.1.1
          enhanced-mode: fake-ip
          fake-ip-range: 198.18.0.1/16
          use-hosts: true
          nameserver:
            # - 192.168.1.1
            - https://dns.alidns.com/dns-query
            - https://doh.pub/dns-query
            - https://2400:3200::1/dns-query
          fallback:
            - tls://13800000000.rubyfish.cn:853
            - https://doh.buzz:8000/dns-query
            - https://doh.beauty:8000/dns-query
            - https://cloudflare-dns.com/dns-query
            - tls://1.1.1.1:853
            - tls://1.0.0.1:853
            - https://1.1.1.1/dns-query
            - https://1.0.0.1/dns-query
            - tls://8.8.8.8:853
            - tls://8.8.4.4:853
            - https://dns.google/dns-query
            - https://dns.twnic.tw/dns-query
          fallback-filter:
            geoip: true
            geoip-code: CN
            ipcidr:
              - 240.0.0.0/4
              - 0.0.0.0/32
          nameserver-policy: null
          listen: 0.0.0.0:53
      prepend-proxy-groups:
        - name: PROXY
          type: select
          proxies:
            - 手动节点
            - 自动节点
            - 故障转移
            - 代理模式

        - name: 手动节点
          type: select

        - name: 自动节点
          type: url-test
          url: http://www.gstatic.com/generate_204
          interval: 86400

        - name: 故障转移
          type: fallback
          url: http://www.gstatic.com/generate_204
          interval: 300

        - name: 代理模式 
          type: select
          proxies:
            - 绕过大陆丨黑名单(GFWlist) # 黑名单模式，意为「只有命中规则的网络流量，才使用代理」
            - 绕过大陆丨白名单(Whitelist) # 白名单模式，意为「没有命中规则的网络流量，统统使用代理」          
            
        - name: 绕过大陆丨黑名单(GFWlist)
          type: url-test
          url: http://www.gstatic.com/generate_204
          interval: 86400
          proxies:
            - DIRECT
            
        - name: 绕过大陆丨白名单(Whitelist)
          type: url-test
          url: http://www.gstatic.com/generate_204
          interval: 86400
          proxies:
            - 故障转移

        - name: Bing 
          type: select
          url: http://www.gstatic.com/generate_204
          interval: 300
          proxies: 
            - 手动节点
            - 自动节点
            - 香港
            - DIRECT

        - name: Microsoft 
          type: select
          url: http://www.gstatic.com/generate_204
          interval: 300
          proxies: 
            - 手动节点
            - 自动节点
            - 香港
            - DIRECT

        - name: Apple 
          type: select
          url: http://www.gstatic.com/generate_204
          interval: 300
          proxies: 
            - 手动节点
            - 自动节点
            - DIRECT

        - name: 流媒体
          type: select
          url: http://www.gstatic.com/generate_204
          interval: 300
          proxies: 
            - PROXY
            - 香港
            - 台湾
            - 日本
            - 新加坡
            - 韩国
            - 美国
            - 其他

        - name: B站
          type: select
          url: http://www.gstatic.com/generate_204
          interval: 300
          proxies: 
            - 手动节点
            - 台湾
            - DIRECT

        - name: 香港 
          type: url-test
          url: http://www.gstatic.com/generate_204
          interval: 300

        
        - name: 台湾 
          type: url-test
          url: http://www.gstatic.com/generate_204
          interval: 300
 
          
        - name: 日本
          type: url-test
          url: http://www.gstatic.com/generate_204
          interval: 300

          
        - name: 新加坡 
          type: url-test
          url: http://www.gstatic.com/generate_204
          interval: 300
 
          
        - name: 韩国 
          type: url-test
          url: http://www.gstatic.com/generate_204
          interval: 300

          
        - name: 美国 
          type: url-test
          url: http://www.gstatic.com/generate_204
          interval: 300

        - name: 其他 
          type: url-test
          url: http://www.gstatic.com/generate_204
          interval: 300

      commands:
        - proxy-groups.手动节点.proxies=[]proxyNames
        - proxy-groups.自动节点.proxies=[]proxyNames|^((?!流量|重置|官网|套餐|订阅).)*$
        - proxy-groups.故障转移.proxies=[]proxyNames|^((?!流量|重置|官网|套餐|订阅).)*$
        - proxy-groups.香港.proxies=[]proxyNames|港|香港|🇭🇰|HK|Hong Kong # 向指定策略组添加订阅中的节点名，可使用正则过滤
        - proxy-groups.台湾.proxies=[]proxyNames|台|台湾|新北|彰化|TW|Taiwan
        - proxy-groups.日本.proxies=[]proxyNames|日|日本|🇯🇵|川日|东京|大阪|泉日|埼玉|沪日|深日|[^-]日|JP|Japan
        - proxy-groups.韩国.proxies=[]proxyNames|韩|韓|韩国|🇰🇷|KR|Korea|KOR|首尔
        - proxy-groups.新加坡.proxies=[]proxyNames|新加坡|🇸🇬|坡|狮城|SG|Singapore
        - proxy-groups.美国.proxies=[]proxyNames|美|美国|🇺🇸|波特兰|达拉斯|俄勒冈|凤凰城|费利蒙|硅谷|拉斯维加斯|洛杉矶|圣何塞|圣克拉拉|西雅图|芝加哥|US|United States
        - proxy-groups.👨其他.proxies=[]proxyNames|^[^香日台新韩美]*$ # 这个可以自定义，这个正则的意思是排除节点名字有“香日台新韩美”的节点。

        # 为各个策略组添加一个DIRECT,避免机场节点无法匹配上面的正则筛选而导致配置出错。应该有其他办法避免，但是我不会。
        - proxy-groups.香港.proxies.0+REJECT
        - proxy-groups.台湾.proxies.0+REJECT
        - proxy-groups.日本.proxies.0+REJECT
        - proxy-groups.韩国.proxies.0+REJECT
        - proxy-groups.新加坡.proxies.0+REJECT
        - proxy-groups.美国.proxies.0+REJECT
        - proxy-groups.其他.proxies.0+REJECT

      prepend-rules:
        #视频网站cdn Start
        - DOMAIN-KEYWORD,.qwedfc,DIRECT
        - DOMAIN-KEYWORD,.fsvod,DIRECT
        - DOMAIN-KEYWORD,ffzyread,DIRECT
        - DOMAIN-KEYWORD,ibytedtos,DIRECT
        - DOMAIN-KEYWORD,puui.qpic,DIRECT  
        - DOMAIN-KEYWORD,puep.qpic,DIRECT
        - DOMAIN-KEYWORD,picnjc.qpic,DIRECT
        - DOMAIN-KEYWORD,snzycdn,DIRECT
        - DOMAIN-KEYWORD,hnts.ymuuy,DIRECT
        - DOMAIN-KEYWORD,yzzy-online,DIRECT
        - DOMAIN-KEYWORD,ffzy-online,DIRECT
        - DOMAIN-KEYWORD,lzcdn,DIRECT
        - DOMAIN-KEYWORD,cdnlz,DIRECT
        - DOMAIN-KEYWORD,monidai,DIRECT
        - DOMAIN-KEYWORD,bdzybf,DIRECT
        - DOMAIN-KEYWORD,vip.,DIRECT
        - DOMAIN-KEYWORD,bfzycdn,DIRECT
        - DOMAIN-SUFFIX,xlzyd.com,DIRECT
        - DOMAIN-SUFFIX,ffzyread.com,DIRECT
        - DOMAIN-SUFFIX,ymuuy.com,DIRECT
        - DOMAIN-SUFFIX,dious.cc,DIRECT
        - DOMAIN-SUFFIX,cjkypo.com,DIRECT
        - DOMAIN-SUFFIX,pstatp.com,DIRECT
        - DOMAIN-SUFFIX,zjtu.tv,DIRECT
        - DOMAIN-SUFFIX,shanshanku.com,DIRECT   
        #视频网站cdn End

        - DOMAIN,login.microsoftonline.com,DIRECT
        - DOMAIN,onedrive.live.com,PROXY
        - DOMAIN-SUFFIX,69shuba.com,PROXY
        - DOMAIN-KEYWORD,xingjiabijichang.com,PROXY
        - DOMAIN-SUFFIX,freeok.vip,DIRECT
        - DOMAIN-SUFFIX,storage.live.com,DIRECT
        - DOMAIN-SUFFIX,skydrive.wns.windows.com,DIRECT
        - DOMAIN-SUFFIX,cnki.net,DIRECT
        - DOMAIN-SUFFIX,ieee.org,DIRECT
        - DOMAIN-SUFFIX,acm.org,DIRECT
        - DOMAIN-SUFFIX,imgtu.com,DIRECT
        - DOMAIN-SUFFIX,dpdk.org,DIRECT
        - DOMAIN-SUFFIX,freenom.com,DIRECT
        - DOMAIN-SUFFIX,sciencedirect.com,DIRECT
        - DOMAIN-SUFFIX,springer.com,DIRECT
        - DOMAIN,clash.razord.top,DIRECT
        - DOMAIN,yacd.haishan.me,DIRECT
        #rule-providers规则
        #注意将需要手动选择的规则集放在前面，规则从上向下执行，否则会被大的proxy等指令集覆盖掉
        - RULE-SET,youtube,流媒体
        - RULE-SET,bing,Bing
        - RULE-SET,microsoft,Microsoft
        - RULE-SET,bilibiliintl,B站
        - RULE-SET,apple,Apple
        - RULE-SET,private,DIRECT
        - RULE-SET,applications,DIRECT
        - RULE-SET,icloud,DIRECT
        - RULE-SET,google,DIRECT
        - RULE-SET,proxy,PROXY
        - RULE-SET,direct,DIRECT
        - RULE-SET,lancidr,DIRECT
        - RULE-SET,cncidr,DIRECT
        - RULE-SET,telegramcidr,PROXY
        - GEOIP,LAN,DIRECT
        - GEOIP,CN,DIRECT
        #兜底
        - MATCH,代理模式

      mix-rule-providers:
      #规则集github地址：https://github.com/Loyalsoldier/clash-rules
      #https://github.com/blackmatrix7/ios_rule_script/tree/master/rule/Clash
        reject:
          type: http
          behavior: domain
          url: https://raw.githubusercontent.com/Loyalsoldier/clash-rules/release/reject.txt
          path: ./ruleset/reject.yaml
          interval: 86400
        icloud:
          type: http
          behavior: domain
          url: https://raw.githubusercontent.com/Loyalsoldier/clash-rules/release/icloud.txt
          path: ./ruleset/icloud.yaml
          interval: 86400
        apple:
          type: http
          behavior: domain
          url: https://raw.githubusercontent.com/Loyalsoldier/clash-rules/release/apple.txt
          path: ./ruleset/apple.yaml
          interval: 86400
        google:
          type: http
          behavior: domain
          url: https://raw.githubusercontent.com/Loyalsoldier/clash-rules/release/google.txt
          path: ./ruleset/google.yaml
          interval: 86400
        proxy:
          type: http
          behavior: domain
          url: https://raw.githubusercontent.com/Loyalsoldier/clash-rules/release/proxy.txt
          path: ./ruleset/proxy.yaml
          interval: 86400
        direct:
          type: http
          behavior: domain
          url: https://raw.githubusercontent.com/Loyalsoldier/clash-rules/release/direct.txt
          path: ./ruleset/direct.yaml
          interval: 86400
        private:
          type: http
          behavior: domain
          url: https://raw.githubusercontent.com/Loyalsoldier/clash-rules/release/private.txt
          path: ./ruleset/private.yaml
          interval: 86400
        gfw:
          type: http
          behavior: domain
          url: https://raw.githubusercontent.com/Loyalsoldier/clash-rules/release/gfw.txt
          path: ./ruleset/gfw.yaml
          interval: 86400
        greatfire:
          type: http
          behavior: domain
          url: https://raw.githubusercontent.com/Loyalsoldier/clash-rules/release/greatfire.txt
          path: ./ruleset/greatfire.yaml
          interval: 86400
        tld-not-cn:
          type: http
          behavior: domain
          url: https://raw.githubusercontent.com/Loyalsoldier/clash-rules/release/tld-not-cn.txt
          path: ./ruleset/tld-not-cn.yaml
          interval: 86400
        telegramcidr:
          type: http
          behavior: ipcidr
          url: https://raw.githubusercontent.com/Loyalsoldier/clash-rules/release/telegramcidr.txt
          path: ./ruleset/telegramcidr.yaml
          interval: 86400
        cncidr:
          type: http
          behavior: ipcidr
          url: https://raw.githubusercontent.com/Loyalsoldier/clash-rules/release/cncidr.txt
          path: ./ruleset/cncidr.yaml
          interval: 86400
        lancidr:
          type: http
          behavior: ipcidr
          url: https://raw.githubusercontent.com/Loyalsoldier/clash-rules/release/lancidr.txt
          path: ./ruleset/lancidr.yaml
          interval: 86400
        applications:
          type: http
          behavior: classical
          url: https://raw.githubusercontent.com/Loyalsoldier/clash-rules/release/applications.txt
          path: ./ruleset/applications.yaml
          interval: 86400
        bing: # Bing的分流规则
          type: http
          behavior: classical
          url: "https://gcore.jsdelivr.net/gh/blackmatrix7/ios_rule_script@master/rule/Clash/Bing/Bing.yaml"
          path: ./ruleset/bing.yaml
          interval: 86400
        microsoft: # microsoft的分流规则
          type: http
          behavior: classical
          url: "https://gcore.jsdelivr.net/gh/blackmatrix7/ios_rule_script@master/rule/Clash/Microsoft/Microsoft.yaml"
          path: ./ruleset/bing.yaml
          interval: 86400
        youtube: # YouTube的分流规则
          type: http
          behavior: classical
          url: "https://gcore.jsdelivr.net/gh/blackmatrix7/ios_rule_script@master/rule/Clash/YouTube/YouTube.yaml"
          path: ./ruleset/youtube.yaml
          interval: 86400    
        bilibiliintl: # BiliBiliIntl的分流规则
          type: http
          behavior: classical
          url: "https://gcore.jsdelivr.net/gh/blackmatrix7/ios_rule_script@master/rule/Clash/BiliBiliIntl/BiliBiliIntl.yaml"
          path: ./ruleset/bilibiliintl.yaml
          interval: 86400

**Comment by VmzXv** [comment, marked as answer]
看到一位老哥自用的，你可以参考下。[地址](https://github.com/abrclano/Self-use-Rules/tree/main/Clash.verge)

**Reply by weineo** [reply]
哈哈感谢，其实之前就自己找到了这个老哥的配置，但是一直没能静得下心仔细看看，粗略看了一眼还以为不能满足我的需求。感谢推荐，能够让我静下心仔细阅读学习，原来是可以达成我的需求的。
